lib/models.js

```diff
@@ -603,11 +679,31 @@ class Cardinality {
   toString() {
     return `${this.min}..${this.isMaxUnbounded ? '*' : this.max}`;
   }
+
+  toJSON() {
+    var out = {
+      "min": this._min,
+      "max": this._max
+    };
+
+    if (this._history && this._history.length > 0) {
+      out["history"] = this._history.map(h => ({
+        "source": h["source"],
+        "min": h._min,
+        "max": h._max
+      }))
```

Comment:
doing the `h => h.toJSON()` approach produces JSON output with underscores, e.g. `"_min":0"`

This is because the `history` object doesn't actually have it's own `toJSON` method, as it's not a custom object.

+1

Reply:
No biggie, but if you find yourself doing this a lot, you could _make_ a `History` class w/ a toJSON function.